### PR TITLE
fix(menu): accessible name with aria-label attribute (VIV-1395)

### PR DESCRIPTION
**Pull request by TaylorJ76**
Adds aria-label to menu.
Also adds Esc key close behaviour.

Adds aria-label to the examples and also the theme switcher menu in docs and fixes it so it can work with keyboard only.

### CODE REVIEW

File: libs/components/src/lib/menu/menu.template.ts [hunk 1]
```diff
@@ -21,6 +21,13 @@ const getClasses = ({
 		['hide-body', items && !(items as unknown as HTMLElement[]).length]
 	);
 
+function handleEscapeKey(menu: Menu, event: Event) {
```
Comment (rachelbt) [inline libs/components/src/lib/menu/menu.template.ts:24]: if the menu is open esc should close it?
If yes - it doesn't work :(
we also have auto dismiss, maybe it suppose to be only there?

**Review by rachelbt — COMMENTED** [review]
all of the examples in menu-item are inside of menu - we should add the aria-label there too

File: libs/components/src/lib/menu/menu.template.ts (hunk 1 above)
Comment (TaylorJ76) [reply libs/components/src/lib/menu/menu.template.ts:24]: You need to be focussed on the menu in order to close the menu with esc.
I will open a story to look at the general keyboard accessibility of menu as there are other issues which as out of the scope of this PR.

File: apps/docs/assets/scripts/theme-handler.js [hunk 2]
```diff
@@ -42,6 +42,11 @@ button.addEventListener('click', toggleMenu);
 const menu = document.querySelector('vwc-menu#dark-mode-menu');
 menu.addEventListener('click', event => setTheme(event.target.text?.toLowerCase()));
 
+menu.addEventListener('keydown', event => {
+    if (event.key === 'Enter' || event.key === 'Space')
+        return setTheme(event.target.text?.toLowerCase());
+});
+
```
Comment (RichardHelm) [inline apps/docs/assets/scripts/theme-handler.js:49]: Should not be necessary to do this
```suggestion
menu.addEventListener('change', event => {
  if (event.target.checked)
    setTheme(event.target.text?.toLowerCase()));
});
```

**Review by RichardHelm — CHANGES_REQUESTED** [review]

File: apps/docs/assets/scripts/theme-handler.js [hunk 3]
```diff
@@ -40,8 +40,10 @@ const button = document.querySelector('vwc-button#dark-mode-toggle');
 button.addEventListener('click', toggleMenu);
 
 const menu = document.querySelector('vwc-menu#dark-mode-menu');
-menu.addEventListener('click', event => setTheme(event.target.text?.toLowerCase()));
-
+menu.addEventListener('change', event => {
+  if (event.target.checked)
+    setTheme(event.target.text?.toLowerCase()));
+});
```
Comment (RichardHelm) [inline apps/docs/assets/scripts/theme-handler.js:46]:
```suggestion
menu.addEventListener('change', event => {
  if (event.target.checked)
    setTheme(event.target.text?.toLowerCase());
});
```

**Review by RichardHelm — APPROVED** [review]

**Pull request merged**